### ISSUE TITLE
[Issue]: Episode images always get overwritten during initial scanning.

### ISSUE THREAD

**Issue by Mihara** [jellyfin/jellyfin#11862]
### Please describe your bug

When episode images (the ones in `metadata`) already exist upon initial scanning, and a complete NFO file also does, they get overwritten from online sources anyway.

This is particularly annoying when the database sources are bogus, since you can't easily replace these images without clicking on them one by one, even if you can trivially re-generate them with a script.

The only way to get around this so far that I found is let the show get scanned completely with bogus episode images, delete the images, generate them again manually, and then "Replace all metadata". Doing it in any other way gets the images overwritten.

### Reproduction Steps

1. Scan a TV show into the library, with NFO writing out enabled, and all images kept together with the show.
2. Once you have a show fully kitted out in metadata, move it out of the library directory.
3. Refresh the library, so that the show is completely gone from it.
4. Move the show back, replace every episode image with a dummy image file.
5. Scan the library for new files again.

Observe every episode image being back to whatever was fetched from the metadata provider anyway.

### Jellyfin Version

10.9.0

### if other:

10.9.3

### Environment

```markdown
- OS: Ubuntu 22.04
- Linux Kernel: 6.8.10-1-liquorix-amd64
- Virtualization: None

Pretty sure most of the rest is not relevant.
```

### Jellyfin logs

```shell
There is no log output that appears relevant, as nothing gets reported regarding *where* were the images acquired.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gnattu** [comment]
If you can, please test https://github.com/jellyfin/jellyfin/pull/11743 and see if that fixed your issue

**Comment by cvium** [comment]
Need more information here. Folder structure, contents of the NFO etc.

**Comment by Mihara** [comment]
This is still happening in 10.9.4, which supposedly contains #11743 - so no, that didn't help.

Assume we're starting with the following directory structure:

```
TV
└── Invincible
    ├── Season 01
    │   ├── metadata
    │   │   ├── Invincible S01E01.jpg
    │   │   └── Invincible S01E02.jpg
    │   ├── Invincible S01E01.mkv
    │   └── Invincible S01E02.mkv
    ├── folder.jpg
    └── tvshow.nfo
```

where tvshow.nfo contains:
```
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<tvshow>
  <imdb_id>tt6741278</imdb_id>
  <tmdbid>95557</tmdbid>
  <tvdbid>368207</tvdbid>
  <tvmazeid>37196</tvmazeid>
</tvshow>
```

And scan this into the database with "Scan for new and updated files."

`folder.jpg` will be retained, but every file in `metadata` will be overwritten with whatever the episode thumbnailing logic supplies. It's like it doesn't even try to check if those already exist.

From there on, they seem to get cached somewhere, because replacing them on disk does not necessarily result in refreshing them. (Sometimes, editing metadata and saving it does result in them refreshing, sometimes not.)

Overwriting files without an explicit checkbox should not happen, should it?

**Comment by Shadowghost** [comment]
This is likely happening because the NFO includes remote image sources. Those are always preferred if an NFO exists. Can you try removing those sources from the NFO and check if the local images are properly picked up?
I'll take a look on how we could prevent this (we'll likely need to modify the NFO when you replace the episode image).

**Comment by Mihara** [comment]
> This is likely happening because the NFO includes remote image sources. 

I find that somewhat unlikely, if only for one reason I have observed already: If the top episode image source is the screenshot grabber, grabbed screenshots will *also* replace pre-existing episode images. Unless "Screen Grabber" also counts as a remote image source, which it internally might.

And if I just remove database IDs from the NFO, the series isn't guaranteed to get picked up at all.

**Comment by Mihara** [comment]
Seems to be fixed in 10.9.5 for me, thank you!
